File: app/src/main/java/app/solocoin/solocoin/api/APIService.java
package app.solocoin.solocoin.api;

import com.google.gson.JsonObject;

import org.json.JSONObject;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.PUT;

public interface APIService {
    @Headers("Content-Type: application/json")
    @POST("callbacks/mobile_login")
    Call<JsonObject> doMobileLogin(@Body JsonObject body);

    @Headers("Content-Type: application/json")
    @POST("callbacks/mobile_sign_up")
    Call<JsonObject> doMobileSignup(@Body JsonObject body);

    @Headers("Content-Type: application/json")
    @GET("user/profile")
    Call<JsonObject> showUserData(@Header("Authorization") String authToken);

    @Headers("Content-Type: application/json")
    @POST("user")
    Call<JsonObject> doUserUpdate(@Body JsonObject body);

    @Headers("Content-Type: application/json")
    @POST("notification_tokens")
    Call<JsonObject> doRegisterNotificationToken(@Body JsonObject body);

    @Headers("Content-Type: application/json")
    @POST("sessions/ping")
    Call<JsonObject> pingSession(@Header("Authorization") String authToken, @Body JsonObject body);

//    @Headers("Content-Type: application/json")
//    @POST("sessions/end")
//    Call<JsonObject> endSession(@Body JsonObject body);
}

File: app/src/main/java/app/solocoin/solocoin/repo/ApiService.kt
package app.solocoin.solocoin.repo

import app.solocoin.solocoin.model.Milestones
import app.solocoin.solocoin.model.Reward
import com.google.gson.JsonObject
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.*

/**
 * Created by Aditya Sonel on 22/04/20.
 */

interface ApiService {
    @Headers("Content-Type: application/json")
    @POST("callbacks/mobile_login")
    suspend fun mobileLogin(@Body body: JsonObject): Response<JsonObject>

    @Headers("Content-Type: application/json")
    @POST("callbacks/mobile_sign_up")
    suspend fun mobileSignUp(@Body body: JsonObject): Response<JsonObject>

    @Headers("Content-Type: application/json")
    @GET("user/profile")
    suspend fun userData(@Header("Authorization") authToken: String): Response<JsonObject>

    @Headers("Content-Type: application/json")
    @PATCH("user")
    suspend fun userUpdate(@Header("Authorization") authToken: String, @Body body: JsonObject): Response<JsonObject>

    @Headers("Content-Type: application/json")
    @POST("notification_tokens")
    suspend fun registerNotificationToken(@Body body: JsonObject): Response<JsonObject>

    @Headers("Content-Type: application/json")
    @POST("sessions/ping")
    fun pingSession(
        @Header("Authorization") authToken: String,
        @Body body: JsonObject
    ): Call<JsonObject>

    @Headers("Content-Type: application/json")
    @GET("user/profile")
    suspend fun getProfile(@Header("Authorization") authToken: String): Response<JsonObject>

    @GET("rewards_sponsors")
    suspend fun getOffers(@Header("Authorization") authToken: String): Response<ArrayList<Reward>>

    @Headers("Content-Type: application/json")
    @POST("user/redeem_rewards")
    suspend fun redeemRewards(
        @Header("Authorization") authToken: String,
        @Body body: JsonObject
    ): Response<JsonObject>

    @GET("user/badges")
    suspend fun getBadgesLevels(@Header("Authorization") authToken: String): Response<Milestones>

    @GET("questions/daily")
    suspend fun getDailyQuiz(@Header("Authorization") authToken: String): Response<JsonObject>

    @GET("questions/weekly")
    suspend fun getWeeklyQuiz(@Header("Authorization") authToken: String): Response<JsonObject>

    @Headers("Content-Type: application/json")
    @POST("user_questions_answers")
    suspend fun submitQuizAnswer(
        @Header("Authorization") authToken: String,
        @Body body: JsonObject
    ): Response<JsonObject>
}